chemprop/data/data.py

```diff
@@ -510,7 +589,7 @@ def normalize_features(self, scaler: StandardScaler = None, replace_nan_token: i
 
         return scaler
 
-    def normalize_targets(self) -> StandardScaler:
+    def normalize_targets(self) -> Union[List[StandardScaler], StandardScaler]:
```

Comment:
Needing to have every location for the scaler variable support both StandardScaler and List[StandardScaler] is going to be very cumbersome. It's probably taking a lot of code to support the ambiguity now, and I expect it will make development around it more confusing in the future. I am not sure that it's necessary.

A single Scaler under molecule-level targets will carry a mean and a stdev for each task. Couldn't we have a single Scaler carry a list of mean and stdev for each task here too? You would need to change how the Scaler processed them, but that seems like the easier way to keep all this consistent. To keep things extra clear, you could create a new class of Scaler. Then StandardScaler and AtomBondScaler (or whatever the name was) could have a common Scaler class that they are subclasses of.

Reply:
I created AtomBondScaler class in scaler.py and normalize_atom_bond_targets() function in MoleculeDataset class. The  AtomBondScaler can carry a mean and a stdev for each task in atom- and bond-level targets. normalize_atom_bond_targets() is created to be distinguished from normalize_targets().